scripts/src/errors.ts

```diff
@@ -7,21 +41,54 @@ export type ErrorType =
 	| 'ServerError'
 	| 'FriendbotError'
 	| 'InvalidAddressError'
-	| 'TransactionFailedError'
 	| 'ChannelBusyError'
+	| 'TransactionFailedError'
 	| 'NetworkMismatchedError'
-	| 'InvalidDataError';
+	| 'InvalidDataError'
+	| 'BadRequestError'
+	| 'InternalError'
+	| 'NoAccountError'
+	| 'AccountExists'
+	| 'LowBalanceError'
+	| 'AccountNotActivatedError'
+	| 'HorizonError'
+	| 'ResourceNotFoundError';
 
 export interface KinSdkError extends Error {
 	readonly type: ErrorType;
 }
 
+export class HorizonError extends Error implements KinSdkError {
+	readonly type: ErrorType = 'HorizonError';
+
+	private readonly _resultTransactionCode?: string;
+	private readonly _resultOperationsCode?: string[];
+	readonly errorCode: number;
+
+	constructor(readonly msg: string, readonly errorBody: ErrorResponse, readonly title?: string) {
+		super(`${msg}, error code: ${errorBody.status} ` + ((title ? `title: ${errorBody.title}` : "")));
+		this.errorCode = errorBody.status;
+		this.errorBody = errorBody;
+		this._resultTransactionCode = ErrorUtils.getTransaction(errorBody);
+		this._resultOperationsCode = ErrorUtils.getOperations(errorBody);
+	}
+
+		public get resultTransactionCode(): string | undefined {
```

Comment:
resultTransaction/Operation should be only on `TransactionFailedError` and then `resultTransactionCode` should not be optional (there's always a value).